app/controllers/maglev/application_controller.rb

```diff
@@ -13,5 +13,9 @@ class ApplicationController < ::ApplicationController
     def use_engine_vite?
       true
     end
+
+    def resource_id
+      Maglev.uuid_as_primary_key? && params[:id] ? params[:id][0..35] : params[:id]
```

Comment:
this is smarter and probably faster than regex matching. nice.

Reply:
thanks! I've added a comment so that we know where 35 comes from.